webapp/components/NewsList/NewsList.vue

```diff
@@ -62,16 +82,103 @@ export default {
     }
   },
   methods: {
-    updateitem (eventitem) {
-      this.items = this.items.map(item => eventitem.id === item.id ? eventitem : item)
+    async updateitem (eventitem) {
+      const upvotePost = gql`
+        mutation upvotePost($id: ID!) {
+          upvote(id: $id ) {
+            id
```

Comment:
If you request the `id` you don't have to update the post yourself :point_down: `apollo-client` is doing it for you.